src/segmentation.rs

```diff
@@ -36,7 +39,7 @@ pub fn segmentation_optimize<T: Pixel>(
     fs.segmentation.data[i][SegLvl::SEG_LVL_ALT_Q as usize] = match i {
       0 => 0,
       1 => TEMPORAL_RDO_QI_DELTA,
-      2 => -TEMPORAL_RDO_QI_DELTA,
+      2 => -TEMPORAL_RDO_QI_DELTA.max(offset_lower_limit),
```

Comment:
Operator precedence here is subtly misleading.
```suggestion
      2 => (-TEMPORAL_RDO_QI_DELTA).max(offset_lower_limit),
```